### DISCUSSION TITLE
Relation manger with pivot model: cast not working on edit

### DISCUSSION THREAD

**Discussion by drcapins** [filamentphp/filament#13618] [Help]
### Package

Panel builder

### Package Version

v3.2.92

### How can we help you?

I create a RelationManager many to many, with pivot field to store.
On attach method, with a repeater, it correctly save a json column in the pivot table.
On edit, when the form method fetch the data from the table, it doesn't cast the json to array, so throw an error:

foreach() argument must be of type array|object, string given, launched on 
vendor/filament/forms/src/Components/Repeater.php :783

the pivot model is:
`class Preiscription extends Pivot
{
    protected $table = "race_tranche_user";

    protected function casts(): array
    {
        return [
            'answers' => 'array',
        ];
    }
}`
but it seems to ignore it on edit. On attach method works.

**Comment by gaetandezeiraud** [comment]
I have the same problem, also with an array. Have you find a solution?

**Reply by sirlionlabs** [reply]
I also have this issue.  I'm extending a RelationManager from FilamentBlog package. Adding a "New" relation works but editing an existing relation does not seem to cast.

EDIT Found my issue: a resource subpage (editResource) was calling the parent Model which did not have the cast for my extended json column.
